packages/terra-functional-testing/src/webpack-server/webpack-server.js

```diff
@@ -73,8 +76,30 @@ class WebpackServer {
         if (stats.hasErrors()) {
           logger.error('Webpack compiled with errors.');
           reject();
-        } else {
+        } else if (!this.gridUrl) {
           resolve();
+        } else {
+          let response = '';
+          http.get(this.gridUrl, (res) => {
+            const { statusCode } = res;
+            if (statusCode >= 200 && statusCode <= 299) {
+              res.on('data', (chunk) => {
+                response += chunk;
+              });
+              res.on('end', () => {
+                if (!res.complete) throw new SevereServiceError(`${this.gridUrl} connection was terminated while the message was still being sent`);
+                if (JSON.parse(response).value.ready) {
+                  resolve();
+                } else {
+                  throw new SevereServiceError(`${this.gridUrl} failed to return a ready response. Check to ensure the selenium grid is stable`);
```

Comment:
I like this message that provides a suggestion to check to make sure the remote selenium grid is stable. For the other failure cases, do we know what may be some of the common causes and provide a suggestion?

For examples:
- For the failure on line 101, is that failure most likely due to not having a valid IP address (WDIO_EXTERNAL_HOST)? If so can we add a suggestion to ensure WDIO_EXTERNAL_HOST is correct/provided.
- For the failure on line 90, when would that occur? Would the failure go away by rerunning the tests again?

Reply:
So for the failure on line 101, that usually a failure when we get a timeout response or a connection refused (so far) which is usually due to vpn issues. Would we want something like 

> "Failed to connect to url ${this.gridUrl}, make sure you have access. Error thrown: ${e.message}

As for line 90 I'm not sure exactly when that would  occur but it was present in most of the examples I was looking at. I think it would more likely to only happen in larger responses that would be broken up into multiple responses by `http` which I think theoretically shouldn't be present in a response from the status page of a gridUrl. I'm not sure if the best option is to just remove it or leave as a just in case bit of defensive coding.

Reply:
Would we get the failure on line 101 if connected to vpn but with an invalid IP address? One common failure is consumers are on the vpn but either didn't provide an IP address or the IP address is invalid. If an invalid IP address can lead to the failure on line 101, it would be helpful to provide a concrete suggestion since consumers may not know what to do/try when they see "make sure you have access".

Something along the line of:

> Failed to connect to url ${this.gridUrl}. Ensure you are connected to the network and/or a valid IP address is provided using WDIO_EXTERNAL_HOST. Error thrown: ${e.message}

I think of these failure messages as a way to inform what went wrong so consumers/we know what to do to attempt to fix it.

Reply:
No we wouldn't get that failure. I haven't been able to find a way to successfully test that particular instance of the WDIO_EXTERNAL_HOST being wrong, that also works on Jenkins.